### PR TITLE
LocalDate: Finances Swapover

### PR DESCRIPTION
**Pull request by Windchild292**
This moves finances to LocalDate, one of the last few systems to do so for.

Finances has fingers all over the product, so this one has a bunch of files changed.

Whitespace differences off, and I'd do the first two commits separately as both are code cleanup, formatting fixes, and light refactoring.

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/finances/Loan.java [hunk 1]
```diff
@@ -93,102 +85,79 @@ public Loan(Money p, int r, int c, int y, int s, GregorianCalendar today, String
...
+        switch (schedule) {
             case Finances.SCHEDULE_BIWEEKLY:
-                if(nextPayment.get(Calendar.DAY_OF_WEEK) == 1) {
-                    if(grace > 0) {
-                        grace--;
-                    }
-                    else {
-                        keepGoing = false;
-                    }
+                temp = 8 - nextPayment.getDayOfWeek().getValue();
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/campaign/finances/Loan.java:101]: This logic needs an explanation, if you're just looking for the next Monday/Sunday or something like that there is a more obvious way:
```
d.with(TemporalAdjusters.next(DayOfWeek.MONDAY));
```

File: MekHQ/src/mekhq/campaign/finances/Loan.java [hunk 2]
```diff
@@ -93,102 +85,79 @@ public Loan(Money p, int r, int c, int y, int s, GregorianCalendar today, String
...
                 break;
             case Finances.SCHEDULE_MONTHLY:
-                if(nextPayment.get(Calendar.DAY_OF_MONTH) == 1) {
-                    if(grace > 0) {
-                        grace--;
-                    }
-                    else {
-                        keepGoing = false;
-                    }
+                temp = nextPayment.getDayOfMonth();
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/campaign/finances/Loan.java:108]: Same as above comment.

File: MekHQ/src/mekhq/campaign/finances/Finances.java [hunk 3]
```diff
@@ -111,37 +109,42 @@ public boolean isInDebt() {
...
-        return Utilities.getDiffFullYears(wentIntoDebt, cal);
     }
 
-    public int getPartialYearsInDebt(GregorianCalendar cal) {
-        if (wentIntoDebt == null) {
-            return 0;
+    public int getPartialYearsInDebt(LocalDate date) {
+        if (wentIntoDebt != null) {
+            Period period = Period.between(wentIntoDebt, date);
+            if ((period.getMonths() > 0) || (period.getDays() > 0)) {
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/campaign/finances/Finances.java:123]: Does this logic match the old logic, the old logic gave you `days % 365 > 0 ? 1 : 0`. Is this equivalent?

File: MekHQ/src/mekhq/gui/dialog/CustomizeScenarioDialog.java [hunk 4]
```diff
@@ -371,44 +370,37 @@ private void btnCloseActionPerformed(java.awt.event.ActionEvent evt) {
...
                     return;
                 } else {
-                    //Calendar math necessitated by variations in locales
-                    GregorianCalendar nextMonday = new GregorianCalendar();
-                    nextMonday.setTime(campaign.getDate());
-                    nextMonday.add(Calendar.DAY_OF_MONTH, 1);
-                    while (nextMonday.get(Calendar.DAY_OF_WEEK) != Calendar.MONDAY) {
-                        nextMonday.add(Calendar.DAY_OF_MONTH, 1);
-                    }
+                    LocalDate nextMonday = campaign.getLocalDate()
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/gui/dialog/CustomizeScenarioDialog.java:384]: Use TemporalAdjusters.

File: MekHQ/src/mekhq/campaign/finances/Finances.java (hunk 3 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/finances/Finances.java:123]: It does. Period in single calculations will split the time between into days, months, and years. If there is no difference in a category, the value will be 0.

File: MekHQ/src/mekhq/campaign/finances/Loan.java [hunk 5]
```diff
@@ -93,102 +87,72 @@ public Loan(Money p, int r, int c, int y, int s, GregorianCalendar today, String
...
-        }
-        while(keepGoing) {
-            nextPayment.add(GregorianCalendar.DAY_OF_YEAR, 1);
-            switch(schedule) {
+
+        // First, we need to increase the number of days by one
+        nextPayment = nextPayment.plusDays(1);
+
+        // Then, we create a variable to hold temporary difference values
+        int temp;
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/campaign/finances/Loan.java:98]: Still needed?

Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/finances/Loan.java:98]: Nope, and odd that my IDE didn't prompt about it being unused.

**Pull request merged**